neurogym/config/components/trigger.py

```diff
@@ -0,0 +1,28 @@
+from typing import Literal
+
+from pydantic import model_validator
+
+from neurogym.config.base import ConfigBase
+
+
+class TriggerConfig(ConfigBase):
+    """Generic configuration for components that run by 'trials' or 'steps'.
+
+    Attributes:
+        trigger: Either 'trials' or 'steps'.
+        value: Positive integer corresponding to the trigger.
+    """
+
+    trigger: Literal["trials", "steps"]
+    value: int
+
+    @model_validator(mode="after")
+    def validate_trigger(self):
+        """Ensure value is positive and trigger is valid."""
+        if self.value <= 0:
```

Comment:
We can probably remove the entire validation method by using Pydantic classes 🙂:

```python
from pydantic import PositiveInt
...
value: PositiveInt = 10
```

I would vote to assign default values to any config variables (also to `trigger`).

Reply:
Nice, I removed the validation and I inserted the PositiveInt. The agent and the monitor do have defaults, while the trigger doesn't (it's never used by itself, only inherited, so it's okay)